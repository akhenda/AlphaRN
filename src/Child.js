import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';

import { capitalizeWords } from './utils/capitalize';

export default class Child extends Component {
  getCapitalizedText() {
    const { text } = this.props;
    if (text) return capitalizeWords(text);

    return 'You have not written anything!';
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>
          Capitalized text:
        </Text>
        <Text style={styles.text}>
          {this.getCapitalizedText()}
        </Text>
        <TouchableOpacity onPress={this.props.onClear}>
          <Text style={styles.button}>
            Clear text
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

Child.propTypes = {
  text: PropTypes.string.isRequired,
  onClear: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    alignItems: 'center',
  },
  title: {
    fontSize: 20,
    marginBottom: 20,
  },
  text: {
    fontSize: 18,
    padding: 20,
  },
  button: {
    marginTop: 30,
    fontSize: 30,
    color: '#007aff',
  },
});
