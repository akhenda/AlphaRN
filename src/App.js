/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TextInput,
} from 'react-native';

import Child from './Child';

const instructions = Platform.select({
  ios: 'Press Cmd+R to reload,\n' +
    'Cmd+D or shake for dev menu',
  android: 'Double tap R on your keyboard to reload,\n' +
    'Shake or press menu button for dev menu',
});

export default class App extends Component<{}> {
  constructor(props) {
    super(props);
    this.state = { text: '' };
  }

  setText(text) {
    this.setState({ text });
  }
  
  handleTextChange(text) {
    this.setText(text);
  }

  clearText() {
    this.setText('');
  }

  render() {
    const { text } = this.state;

    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit App.js
        </Text>
        <Text style={styles.instructions}>
          {instructions}
        </Text>
        <TextInput
          value={text}
          style={styles.input}
          placeholder={'Write something...'}
          onChangeText={txt => this.handleTextChange(txt)}
        />
        <Child text={text} onClear={() => this.clearText()} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  input: {
    width: '80%',
    height: 60,
    padding: 10,
    alignSelf: 'center',
  },
});
