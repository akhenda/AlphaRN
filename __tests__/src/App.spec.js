import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from 'src/App';
import Child from 'src/Child';


describe('App', () => {
  let component;
  let textInput;
  const defaultState = { text: '' };

  test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  beforeEach(() => {
    component = shallow(<App />);
    textInput = component.find('TextInput');
  });

  it('has default state', () => {
    expect(component.state()).toEqual(defaultState);
  });

  it('renders welcome text', () => {
    const expectedText = 'Welcome to React Native!';
    const text = component.find('Text').first().children().text();
    expect(text).toEqual(expectedText);
  });

  it('renders input field with placeholder', () => {
    const expectedPlaceholder = 'Write something...';
    expect(textInput.length).toBe(1);
    expect(textInput.props().placeholder).toEqual(expectedPlaceholder);
  });

  describe('when text changes', () => {
    const newText = 'random string';

    beforeEach(() => {
      textInput.simulate('changeText', newText);
    });

    it('updates component state', () => {
      expect(component.state().text).toEqual(newText);
    });

    it('passes text from state to Child', () => {
      const childComponent = component.find(Child);
      expect(childComponent.props().text).toEqual(newText);
    });

    describe('when clearText callback is called', () => {
      beforeEach(() => {
        const childComponent = component.find(Child);
        childComponent.simulate('clear');
      });

      it('resets state', () => {
        expect(component.state()).toEqual(defaultState);
      });
    });
  });
});
