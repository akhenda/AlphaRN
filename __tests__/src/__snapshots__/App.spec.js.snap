// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit App.js
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Write something..."
    style={
      Object {
        "alignSelf": "center",
        "height": 60,
        "padding": 10,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 20,
        }
      }
    >
      Capitalized text:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "padding": 20,
        }
      }
    >
      You have not written anything!
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#007aff",
            "fontSize": 30,
            "marginTop": 30,
          }
        }
      >
        Clear text
      </Text>
    </View>
  </View>
</View>
`;
